contracts/proxy/utils/Initializable.sol

```diff
@@ -47,19 +70,21 @@ abstract contract Initializable {
      * @dev Modifier to protect an initializer function from being invoked twice.
      */
     modifier initializer() {
-        // If the contract is initializing we ignore whether _initialized is set in order to support multiple
-        // inheritance patterns, but we only do this in the context of a constructor, because in other contexts the
-        // contract may have been reentered.
-        require(_initializing ? _isConstructor() : !_initialized, "Initializable: contract is already initialized");
-
-        bool isTopLevelCall = !_initializing;
+        bool isTopLevelCall = _setInitializedVersion(1);
```

Comment:
After re-reviewing the code I don't think it makes sense to set `_initializing` as part of `_setInitializedVersion`. I know we did this to optimize bytecode size but I think for now we should move `_initializing = true` outside of the function, and we can consider optimizing it later and do benchmarks.